src/components/BlogArticlePreview/BlogArticlePreview.js

```diff
@@ -1,37 +1,78 @@
-import React from 'react';
-import cx from 'classnames';
-import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import HttpService from '../../service/HttpService/httpService';
 
 import './BlogArticlePreview.scss';
 
-export const CN = 'article-preview';
+export const CN = 'blog';
 
-const BlogArticlePreview = (props) => {
-  const { imageSrc, heading, text } = props;
-  const backgroundImageURL = {
-    backgroundImage: `url(${imageSrc})`
-  };
+class BlogArticlePreview extends Component {
+  constructor(props) {
+    super(props);
 
-  return (
-    <div className={cx(CN)} style={backgroundImageURL}>
-      <div className={`${CN}__content`}>
-        <h2 className={`${CN}__heading`}>{heading}</h2>
-        <p className={`${CN}__text`}>{text}</p>
+    this.state = {
+      blogsList: []
+    };
+  }
+
+  componentDidMount() {
+    this.getBlogs();
+  }
+
+  async getBlogs() {
+    const userAPI = new HttpService();
+
+    try {
+      const response = await userAPI.get(`${process.env.BASE_URL}/blogs`);
+      if (response && response.data) {
+        this.setState({ blogsList: response.data });
+      }
+    } catch (error) {
+      throw (new Error());
+    }
+  }
+
+  renderBrands(blogsList) {
```

Comment:
I just dont get it!
If you have value in state, why you are passing it as parameter?

Reply:
in order not to get it out of state 2 times

Reply:
it should be taken out if state on line 35, 59

Reply:
You can read the state as much as you want. It doesn't trigger a rerender